subgraphs/euler-finance/src/mappings/handlers.ts

```diff
@@ -139,19 +247,24 @@ function queryEulerGeneralView(marketIds: string[], block: ethereum.Block): Eule
 }
 
 // initiates market / protocol updates in syncWithEulerGeneralView()
-function updateProtocolAndMarkets(block: ethereum.Block): void {
-  let blockNumber = block.number.toI32();
+function updateProtocolAndMarkets(event: ethereum.Event, marketId: string): void {
+  let blockNumber = event.block.number.toI32();
   let protocolUtility = getOrCreateProtocolUtility(blockNumber);
   let markets = protocolUtility.markets;
+  const marketIndex = markets.indexOf(marketId);
+  assert(marketIndex >= 0, `[updateProtocolAndMarkets]marketId=${marketId} not in protocolUtility.markets`);
+  markets = [marketId, USDC_ERC20_ADDRESS];
```

Comment:
Because the original code use a different way to compute price based on whether USDC is included in the query (https://github.com/messari/subgraphs/blob/e130ec9f338b89dd99e894c90f39f55c55c2856e/subgraphs/euler-finance/src/mappings/helpers.ts#L305)

Reply:
I understand it now, resolved